src/jit/function-builder.cc

```diff
@@ -195,14 +197,39 @@ bool FunctionBuilder::buildIL() {
   return true;
 }
 
-void FunctionBuilder::SetUpLocals(TR::IlBuilder* b, const uint8_t** pc, VirtualStack* stack) {
+bool FunctionBuilder::SetUpLocals(TR::IlBuilder* b, const uint8_t** pc, VirtualStack* stack) {
   // Add placeholders onto the virtual stack. These values should never be actually read, but will
   // eventually be dropped using drop or drop_keep in the function epilogue.
   for (size_t i = 0; i < fn_->param_and_local_types.size(); i++) {
-    stack->Push(nullptr);
+    auto t = fn_->param_and_local_types.at(i);
+    auto locals_offset = fn_->param_and_local_types.size() - fn_->local_count;
+
+    if (i < locals_offset) {
+      auto param_addr = PickPhys(b, locals_offset - i);
+      auto param = b->LoadIndirect("Value", TypeFieldName(t), param_addr);
```

Comment:
You need to verify that `t` is a supported type here as well, otherwise `TypeFieldName` will assert when using an unsupported type as a parameter.

Reply:
Should be fixed now.